### DISCUSSION TITLE
Modality Worklist Entry Creation

### DISCUSSION THREAD

**Discussion by EneaSingh** [dcm4che/dcm4che#904] [Q&A] [answered]
Hello, pardon for asking such a question but could you provide some clear directions on how I could implement the following functionality (I mean, which packs to download, in what package can I find the right classes, what are the components..):

- Need to run a DICOM server
- Need to create/add MWL Entries to the server

Many thanks in advance & kind regards,
Enea

**Comment by gunterze** [comment, marked as answer]
You may run [dcm4chee-arc-5.x](https://github.com/dcm4che/dcm4chee-arc-light/wiki), just using its MWL SCP - you may disable all other Transfer Capabilities enabled by default configuration - and use its RESTful service: [POST /dcm4chee-arc/aets​/{aet}​/rs​/mwlitems](https://petstore.swagger.io/index.html?url=https://raw.githubusercontent.com/dcm4che/dcm4chee-arc-light/master/dcm4chee-arc-ui2/src/swagger/openapi.json#/MWL-RS/CreateUpdateSPS) to create or update MWL Entries managed by the server.

**Reply by EneaSingh** [reply]
Thanks a lot Gunter.
Oh wait, you mean i don't need the source code?

Is it correct to say these are the parts I need (I just need the minimum for the functionality):

- Binaries for MySQL
- MySQL Database
- I need to install Wildfly?
- I must have LDAP or is it optional?
- I must have KeyCloak if I want to run the RESTful Service? If I don't follow this (blue square) there is no REST Service?

![afbeelding](https://user-images.githubusercontent.com/79855869/109632983-e59b8080-7b47-11eb-8881-8692e0634cc7.png)

- Any other components?

Many thanks in advance,
Enea

**Reply by gunterze** [reply]
Simplest way to install the archive - particularly on a Linux box - is [running it on docker](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Running-on-Docker).

**Reply by EneaSingh** [reply]
Thank you but I still have the questions:

- Is LDAP mandatory?
- Is KeyCloak mandatory?

Regards,
Enea

**Reply by vrindanayak** [reply]
Yes, LDAP is mandatory as all the archive configurations are in LDAP. Keycloak is mandatory only if you want to run secured version of archive, not otherwise. See [Run secured archive services on a single host](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Run-secured-archive-services-on-a-single-host)
